### ISSUE TITLE
Update status in the checkpoint when training hit the last epoch/episode

### ISSUE THREAD

**Issue by tkornut** [IBM/mi-prometheus#85]
In the case when:
 - model is not better than previous/saved model AND
 - status changes (which will happen ONLY when we reached the limits for episode or epoch)

Then we should:
 - load data from checkpoint
 - update status to the current status, leaving the model unchanged (?)
 - save the updated checkpoint

This requires to additionally save training status along with loss in model.

**Comment by vmarois** [comment]
Yes :+1: 
A case where this is useful is when the **best_model** (based on the validation loss criterion) is saved *early* during the training but it **never** went under the loss threshold, thus did not actually "converged". In this case, the information **model_saved_timestamp** and **terminal_status_update_timestamp** will help the user understand this case

**Comment by tkornut** [comment]
List of possible statuses (for now):
 - Not converged (Episode Limit reached)
 - Not converged (Epoch Limit reached)
 - Not converged (Interrupted by the user)
 - Converged (Full Validation Loss went below Loss Stop threshold)
 - Converged (Partial Validation Loss went below Loss Stop threshold)
